aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,689 @@
+
+
+``@inject``
+^^^^^^^^^^^^^^
+The ``@inject`` directive enables you to inject a service from your :doc:`service container </fundamentals/dependency-injection>`  into your Razor page for use. See :doc:`/mvc/views/dependency-injection`.
+
+An additional feature of ``@inject`` is that it also enables you to replace a few properties that are automatically injected into a Razor page. For instance, we could replace the Html property on our Razor page with the hosting environment:
+
+.. literalinclude:: razor/sample/Views/Home/Contact4.cshtml
+  :language: html
+
+If you inject a property that already exists on your Razor page and has been ``@injected`` before, or is one of the following properties:
+
+- Html
+- Json
+- Component
+- Url
+- :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.ModelExpressionProvider`
+
+Your ``@inject`` statement will override that property and it will be available as the overridden type.
+
+``@functions``
+^^^^^^^^^^^^^^
+
+The ``@functions`` directive enables you to add function level content to your Razor page. The syntax is:
+
+.. code-block:: none
+
+  @functions { // C# Code }
+
+For example:
+
+.. literalinclude:: razor/sample/Views/Home/Contact6.cshtml
+  :language: html
+
+Generates the following HTML markup:
+
+.. code-block:: none
+
+  <div>From method: Hello</div>
+
+The generated Razor C# looks like:
+
+.. literalinclude:: razor/sample/Classes/Views_Home_Test_cshtml.cs
+  :language: c#
+  :lines: 1-19
+
+``@section``
+^^^^^^^^^^^^^^
+
+The ``@section`` directive is used in conjunction with the :doc:`layout page </mvc/views/layout>` to enable views to render content in different parts of the rendered HTML page. See :ref:`layout-sections-label` for more information.
+
+
+TagHelpers
+-----------
+
+The following :doc:`/mvc/views/tag-helpers/index` directives are detailed in the links provided.
+
+- :ref:`@addTagHelper <add-helper-label>`
+- :ref:`@removeTagHelper <remove-razor-directives-label>`
+- :ref:`@tagHelperPrefix <prefix-razor-directives-label>`
+
+.. _razor-reserved-keywords-label:
+
+Razor reserved keywords
+------------------------
+
+Razor keywords
+^^^^^^^^^^^^^^^
+
+- functions
+- inherits
+- model
+- section
+- helper   (Not supported by ASP.NET Core.)
+
+Razor keywords can be escaped with ``@(Razor Keyword)``, for example ``@(functions)``. See the complete sample below.
+
+C# Razor keywords
+^^^^^^^^^^^^^^^^^^
+
+- case
+- do
+- default
+- for
+- foreach
+- if
+- lock
+- switch
+- try
+- using
+- while
+
+C# Razor keywords need to be double escaped with ``@(@C# Razor Keyword)``, for example ``@(@case)``. The first ``@`` escapes the Razor parser, the second ``@`` escapes the C# parser. See the complete sample below.
+
+Reserved keywords not used by Razor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- namespace
+- class
+
+The following sample show all the Razor reserved words escaped:
+
+.. literalinclude:: razor/sample/Views/Home/Contact5.cshtml
+  :language: html
+
+Miscellaneous
```

Comment:
How about we remove this section and add the following at the end of the explicit razor expression section:

You can use an explicit expression to concatenate text with an expression result:

```
@{
    var joe = new Person("Joe", 33);
 }

<p>Age@(joe.Age)</p>
```

Without the explicit expression (`<p>Age@joe.Age</p>`) the content would be treated as an email address.